examples/plot_power.py

```diff
@@ -1,86 +1,120 @@
+                                       fmin=fmin, fmax=fmax,
+                                       method=power_method,
+                                       is_epoched=is_epoched)
 
-    main_workflow = pe.Workflow(name=power_analysis_name)
-    main_workflow.base_dir = data_path
+###############################################################################
+# We then connect the nodes two at a time. First, we connect the two outputs
+# (subject_id and session_id) of the infosource node to the datasource node.
+# So, these two nodes taken together can grab data.
 
-    # info source
-    infosource = create_infosource()
+main_workflow.connect(infosource, 'subject_id', datasource, 'subject_id')
+main_workflow.connect(infosource, 'session_id', datasource, 'session_id')
 
-    # data source
-    datasource = create_datasource()
+###############################################################################
+# Similarly, for the inputnode of the power_workflow. Things will become
+# clearer in a moment when we plot the graph of the workflow.
 
-    main_workflow.connect(infosource, 'subject_id', datasource, 'subject_id')
-    main_workflow.connect(infosource, 'sess_index', datasource, 'sess_index')
+main_workflow.connect(datasource, 'raw_file',
+                      power_workflow, 'inputnode.fif_file')
 
-    power_workflow = create_pipeline_power(data_path, freq_bands,
-                                           fmin=fmin, fmax=fmax,
-                                           method=power_method,
-                                           is_epoched=is_epoched)
 
-    main_workflow.connect(datasource, 'raw_file',
-                          power_workflow, 'inputnode.fif_file')
+###############################################################################
+# To do so, we first write the workflow graph (optional)
 
-    return main_workflow
+main_workflow.write_graph(graph2use='colored')  # colored
 
+###############################################################################
+# and visualize it. Take a moment to pause and notice how the connections
+# here correspond to how we connected the nodes.
 
-if __name__ == '__main__':
+from scipy.misc import imread # noqa
+import matplotlib.pyplot as plt # noqa
+img = plt.imread(op.join(data_path, 'power_pipeline', 'graph.png'))
+plt.figure(figsize=(8, 8))
+plt.imshow(img)
+plt.axis('off')
 
-    # run pipeline:
-    main_workflow = create_main_workflow_power()
+###############################################################################
+# Finally, we are now ready to execute our workflow.
 
-    main_workflow.write_graph(graph2use='colored')  # colored
-    main_workflow.config['execution'] = {'remove_unnecessary_outputs': 'false'}
+main_workflow.config['execution'] = {'remove_unnecessary_outputs': 'false'}
 
-    main_workflow.run(plugin='MultiProc', plugin_args={'n_procs': 3})
+# Run workflow locally on 3 CPUs
+main_workflow.run(plugin='MultiProc', plugin_args={'n_procs': 3})
```

Comment:
we should probably do `'n_procs': 1` by default. Not everyone has 3 cores on their laptops.

Reply:
ok!